src/app/pages/applications/chart-releases/chart-releases.component.ts

```diff
@@ -376,39 +359,20 @@ export class ChartReleasesComponent implements AfterViewInit, OnInit, OnDestroy
     });
   }
 
-  rollback(name: string): void {
-    this.rollbackChartName = name;
-    this.dialogService.dialogForm(this.rollBackChart, true);
-    const rollBackList = Object.keys(this.chartItems[this.rollbackChartName].history);
-    const rollBackConfig = this.rollBackChart.fieldConfig[0] as FormSelectConfig;
-    rollBackConfig.value = rollBackList[0];
-    rollBackConfig.options = rollBackList.map((item) => ({
-      label: item,
-      value: item,
-    }));
-  }
-
-  doRollback(entityDialog: EntityDialogComponent): void {
-    const form = entityDialog.formGroup.controls;
-    const payload = {
-      item_version: form['item_version'].value,
-      rollback_snapshot: form['rollback_snapshot'].value,
-    };
-    this.dialogRef = this.mdDialog.open(EntityJobComponent, {
-      data: {
-        title: helptext.charts.rollback_dialog.job,
-      },
-    });
-    this.dialogRef.componentInstance.setCall('chart.release.rollback', [this.rollbackChartName, payload]);
-    this.dialogRef.componentInstance.submit();
-    this.dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
-      this.dialogService.closeAllDialogs();
-      this.refreshChartReleases();
-    });
-    this.dialogRef.componentInstance.failure.pipe(untilDestroyed(this)).subscribe((error) => {
-      this.dialogService.closeAllDialogs();
-      new EntityUtils().handleWsError(this, error, this.dialogService);
-    });
+  onRollback(name: string): void {
+    const chartRelease = this.chartItems[name];
+    this.mdDialog.open(ChartRollbackModalComponent, {
+      data: chartRelease,
+    })
+      .afterClosed()
+      .pipe(untilDestroyed(this))
+      .subscribe((wasRolledBack) => {
+        if (!wasRolledBack) {
```

Comment:
we can use RxJs `filter((wasRolledBack) => !!wasRolledBack)` 
and in .subscribe just use 
`this.refreshChartReleases();` 
without any checks

Reply:
Yeah, I was going to and used `filter(Boolean)` in the past, but then I thought have it may be better to be more explicit.
With `filter(Boolean)` it's easy to miss it and not understand what's going on, with explicit variable it's clear what's happening.